src/core/qgsunittypes.cpp

```diff
@@ -864,6 +979,12 @@ QgsUnitTypes::AreaUnit QgsUnitTypes::distanceToAreaUnit( DistanceUnit distanceUn
     case DistanceKilometers:
       return AreaSquareKilometers;
 
+    case DistanceCentimeters:
```

Comment:
This isn't right... I think you'll need to unfortunately add AreaSquareMillimeters/AreaSquareCentimeters to correctly handle this.